gh-139741: Make `dist-pdf` docs archive build work for macOS

Naitreey opened the pull request:
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNNN: Summary of the changes made
```

Where: gh-NNNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNNNN)
```

Where: [X.Y] is the branch name, for example: [3.13].

GH-NNNNNN refers to the PR number from `main`.

-->

Building docs archive in pdf format on macOS is currently unsuccessful. This PR fixes it.

Changes:

- Remove `--output-sync` on macOS, as macOS's `make` does not yet support the output sync feature.
- macOS doesn't have `nproc` by default; use `getconf _NPROCESSORS_ONLN` instead.

Thanks @m-aciek for providing the initial solution and PR suggestion.

<!-- gh-issue-number: gh-139741 -->
* Issue: gh-139741
<!-- /gh-issue-number -->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--140837.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

Naitreey committed:
Make ``dist-pdf`` docs archive build work for macOS

m-aciek commented on Doc/Makefile:
```diff
@@ -241,7 +241,11 @@ dist-pdf:
 	-sed -i 's/: all-$$(FMT)/:/' build/latex/Makefile
-	(cd build/latex; $(MAKE) clean && $(MAKE) --jobs=$$((`nproc`+1)) --output-sync LATEXMKOPTS='-quiet' all-pdf && $(MAKE) FMT=pdf zip bz2)
+	if [ -n "$(filter output-sync,$(value .FEATURES))" ]; then \
+		(cd build/latex; $(MAKE) clean && $(MAKE) --jobs=$$((`nproc`+1)) --output-sync LATEXMKOPTS='-quiet' all-pdf && $(MAKE) FMT=pdf zip bz2) \
```
I wonder if conditioning use of `nproc` on availability of `output-sync` feature isn't too fragile. E.g. it will fail if MacOS ever updates the Makefile it vendors. Maybe let's switch to `getconf _NPROCESSORS_ONLN` everywhere?

https://stackoverflow.com/a/64571679/1161025

Naitreey replied:
You are right. I've made the suggested change.

m-aciek reviewed (commented):
Confirming it works on MacOS 26.0.1, thank you, sharing one concern

Naitreey committed:
Replace ``nproc`` with ``getconf _NPROCESSORS_ONLN``

``getconf _NPROCESSORS_ONLN`` is available in both Linux and macOS, but
``nproc`` is not. This change prevents the build from breaking again if
macOS ever upgrades its `make` to version 4 or above.

hugovk commented on Doc/Makefile:
```diff
@@ -241,7 +241,11 @@ dist-pdf:
+	if [ -n "$(filter output-sync,$(value .FEATURES))" ]; then \
+		(cd build/latex; $(MAKE) clean && $(MAKE) --jobs=$$((`getconf _NPROCESSORS_ONLN`+1)) --output-sync LATEXMKOPTS='-quiet' all-pdf && $(MAKE) FMT=pdf zip bz2) \
+	else \
+		(cd build/latex; $(MAKE) clean && $(MAKE) --jobs=$$((`getconf _NPROCESSORS_ONLN`+1)) LATEXMKOPTS='-quiet' all-pdf && $(MAKE) FMT=pdf zip bz2) \
```
It's not obvious, but these two lines are identical, except one uses `--output-sync`.

Can we refactor so when it's available, we put "--output-sync" in a variable, and when it's not, put an empty string there?

And then we can have just a single line using that variable.

Naitreey replied:
I have refactored the code. Please take another look.

Naitreey committed:
Refactor code to eliminate duplication

hugovk reviewed (approved):
Thanks, confirmed this is working* on macOS.

\* I don't have latexmk installed, but the commands are being called as expected.
